Fix MOI used in recovery, windup, and kA calcs.

brettle opened the pull request:
Specifically for recovery and windup time:
- Multiply torque by motor ratio. Note that doing this, combined with the existing division of the freespeed by the motor ratio, is equivalent to dividing the moi by the motor ratio squared.
- Update tests to reflect new expected results.
- Calculate windup time as recovery time from a speed of zero to eliminate duplication of calculation.
- Rename a couple variable for clarity.

Specifically for kA:
- Adjust MOI by the motor ratio squared to get MOI seen by motor.
- Divide that MOI by square of radius of shooter instead of flywheel since the surface acceleration of the shooter is  what the kA will be for.
- Don't multiply the motor torque by the motor ratio since that is now addressed by using the modified MOI.

Also remove duplicate calculations of total moi.

brettle committed:
Fix MOI used in recovery and windup time calcs.

Specifically:

Multiply torque by motor ratio. Note that doing this, combined with the
existing division of the freespeed by the motor ratio, is equivalent to
dividing the moi by the motor ratio squared.

Update tests to reflect new expected results.

Calculate windup time as recovery time from a speed of zero to eliminate
duplication of calculation.

Rename a couple variable for clarity.

tervay commented:
Deploy preview for _recalc_ ready!

✅ Preview
https://recalc-3ngdjv2qg-tervay.vercel.app

Built with commit 91a31ac04e2bc4a07954cd5b012b02728e3e2e9f.
This pull request is being automatically deployed with [vercel-action](https://github.com/marketplace/actions/vercel-action)

brettle committed:
Remove duplicate calculations of total moi.

brettle committed:
Fix calculation of Ka.

Specifically:

Adjust MOI by the motor ratio squared to get MOI seen by motor.

Divide that MOI by square of radius of shooter instead of flywheel since
the surface acceleration of the shooter is  what the kA will be for.

Don't multiply the motor torque by the motor ratio since that is now
addressed by using the modified MOI.

brettle commented:
Is there a reason this was closed without merging? For more details on the math, see https://www.chiefdelphi.com/t/new-flywheel-shooter-analysis/439111.
